blocklib/picoscope4000a/picoscope4000a.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef GR_PICOSCOPE4000A_PICOSCOPE4000A_HPP
+#define GR_PICOSCOPE4000A_PICOSCOPE4000A_HPP
+
+#include <picoscope.hpp>
+
+namespace gr::picoscope4000a {
+
+struct Picoscope4000a : public gr::picoscope::Picoscope<Picoscope4000a> {
+    using AnalogPort = fair::graph::PortOut<float>;
+
+    AnalogPort values0;
+    AnalogPort errors0;
+    AnalogPort values1;
+    AnalogPort errors1;
+    AnalogPort values2;
+    AnalogPort errors2;
+    AnalogPort values3;
+    AnalogPort errors3;
+    AnalogPort values4;
+    AnalogPort errors4;
+    AnalogPort values5;
+    AnalogPort errors5;
+    AnalogPort values6;
+    AnalogPort errors6;
+    AnalogPort values7;
+    AnalogPort errors7;
+
```

Comment:
We also need a digital `bool` streaming output port to indicate the detected edge triggers (+ attached tags)

Reply:
Would that be the digital ports of the picoscope device (which the PS4000A doesn't have), or an additional one?

Reply:
For the PS4000A this would be an additional port where the selected logic state of the trigger port (A0...A1)  is mapped to.

For the other models this would be the regular trigger and/or DIO port.

Reply:
trigger logic port tracked as #127